compiler/src/dmd/statementsem.d

```diff
@@ -3476,6 +3465,11 @@ Statement statementSemanticVisit(Statement s, Scope* sc)
             sc.sbreak = null;
             sc.scontinue = null;
         }
+        if (!global.params.useExceptions && oss.tok == TOK.onScopeFailure)
+        {
+            result = new CompoundStatement(oss.loc).statementSemantic(sc);
+            return;
+        }
```

Comment:
`scope(failure)` and `scope(success)` currently use a `catch(Throwable)` cause to detect the error case. Maybe we could weaken them to `catch(Exception)` - in general, or just with `-nothrow`?